robottelo/ui/base.py

```diff
@@ -130,6 +130,31 @@ def search(self, element_name):
             time.sleep(1)
         return None
 
+    def create_a_bookmark(self, name=None, query=None, public=None,
+                          searchbox_query=None):
+        """Bookmark a search on current entity page"""
+        self.navigate_to_entity()
+        prefix = 'kt_' if self.is_katello else ''
+        searchbox = self.wait_until_element(
+            common_locators[prefix + 'search'],
+            timeout=self.button_timeout
+        )
+        if searchbox is None:
+            raise UINoSuchElementError('Search box not found.')
+        if searchbox_query is not None:
+            searchbox.clear()
+            searchbox.send_keys(u'{0}'.format(escape_search(query)))
+        self.click(common_locators['search_dropdown'])
```

Comment:
Yes. They have different xpaths, of course, but basically we have even 2 different forms for katello and non-katello bookmarks, with different elements and different xpaths for everything :). I've spent some time making xpaths which are working for both katello and non-katello, so in the end from robottelo user perspective it looks like there's only one approach and no need to remember type of any entity.